### PR TITLE
Abstract base configuration

### PR DESCRIPTION
**Pull request by pope1838**

### CODE REVIEW

**Review by looseale — APPROVED** [review]

File: shared/src/main/scala/amf/client/remod/AMFGraphConfiguration.scala [hunk 1]
```diff
@@ -87,39 +130,60 @@ private[amf] class AMFGraphConfiguration(
...
-        Set.empty,
-        AMFOptions.default()
-    )
-  }
+  protected def _withRenderOptions[T](renderOptions: RenderOptions): T =
+    copy(options = options.copy(renderOptions = renderOptions)).asInstanceOf[T]
 
-  def predefined(): AMFGraphConfiguration =
-    empty()
-      .withPlugins(List(AMFGraphParsePlugin, AMFGraphRenderPlugin))
```
Comment (AgustinBettati) [inline shared/src/main/scala/amf/client/remod/AMFGraphConfiguration.scala:113]: new predefined config does not set graph rendering and parsing plugins as well as basic resolution pipeline, was this change intentional?

Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/AMFGraphConfiguration.scala:113]: Claramente fue sin querer

**Review by AgustinBettati — APPROVED** [review]
